### PR TITLE
[#6159] Add and Update Docstring for existing functions

### LINKED ISSUE #6159: Chained actions, auth functions and helpers are not documented
**CKAN version**
2.7, 2,8, 2.9, master

**Describe the bug**

Chained actions are missing docstring completely:
https://github.com/ckan/ckan/blob/e89ca125dc68ddb9fe9bad68a401404146ba90c7/ckan/logic/__init__.py#L334-L336
There is a mention of chained actions within get_actions:
https://github.com/ckan/ckan/blob/e89ca125dc68ddb9fe9bad68a401404146ba90c7/ckan/plugins/interfaces.py#L922-L935

Chained auth functions have minimal docstring:
https://github.com/ckan/ckan/blob/e89ca125dc68ddb9fe9bad68a401404146ba90c7/ckan/logic/__init__.py#L634-L639

Chained helpers have minimal docstring:
https://github.com/ckan/ckan/blob/e89ca125dc68ddb9fe9bad68a401404146ba90c7/ckan/lib/helpers.py#L141-L145

Changelogs have minor mentions to them. `chained_action` was added in 2.7 https://github.com/ckan/ckan/pull/3494 and 2.9 docs have a mention to them:
image

`chained_auth_function` was added in 2.8 https://github.com/ckan/ckan/pull/3995 and docs have a small mention of them:
image

`chained_helper` is only in master https://github.com/ckan/ckan/pull/5337 and are not mentioned in the changelog. 

There is a minor example how to use these in:
https://github.com/ckan/ckan/blob/e89ca125dc68ddb9fe9bad68a401404146ba90c7/ckanext/chained_functions/plugin.py#L32-L46

### PR DESCRIPTION
**Pull request by steveoni**
Fixes #6159 

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/lib/helpers.py [hunk 1]
```diff
@@ -139,8 +139,35 @@ def _is_chained_helper(func):
...
+    returning different values and/or raising different exceptions
+    to the caller.
+
+    Usage::
+
+        from ckan.plugins.toolkit import chained_helper
+
+        @chained_helper
+        def ckan_version(next_func, **kw):
+            return next_func(**kw)
```
Comment (amercader) [inline ckan/lib/helpers.py:162]: Perhaps is better to show an example of a function that supports parameters, and be explicit about what those are:

```suggestion
        def check_access(next_func, action, data_dict=None):
        
            return next_func(action, data_dict)
```

File: ckan/logic/__init__.py [hunk 2]
```diff
@@ -332,6 +332,33 @@ def clear_actions_cache():
 
 
 def chained_action(func):
+    '''Decorator function allowing action function to be chained.
+
+    This allows a plugin to modify the behaviour of an existing action
+    function. Chain action function must be defined as
```
Comment (amercader) [inline ckan/logic/__init__.py:338]:
```suggestion
    function. A chained action function must be defined as
```

**Review by amercader — COMMENTED** [review]
This is fantastic and will make life easier for many extension developers @steveoni . I just added two minor suggestions

**Comment by steveoni** [comment]
I've updated the docstring. 
@amercader

**Pull request merged**